### PR TITLE
fix(cli): properly process configFile if provided

### PR DESCRIPTION
**Pull request by Hotell**
**Summary**

while configFile cli was implemented it doesn't work (https://github.com/callstack/linaria/pull/336).

This PR makes it work (processes the actual config file if present).

**Before:**
image

**After:**

image

> Btw using flow was really a PITA from a contributor experience 👀🤔

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -66,6 +66,11 @@ type Options = {
 function processFiles(files: string[], options: Options) {
   let count = 0;
 
+  const configFileOptions = options.configFile
+    ? // $FlowFixMe
+      require(options.configFile)
```
Comment (satya164) [inline src/cli.js]: The config file can be a lot of different formats. The reason it's not working in your case is that seems we are not respecting the config file in `transform.js`.

So we need to add the following:

```js
const explorer = cosmiconfig('linaria');

...

const { configFile, ...rest } = options.pluginOptions;
  // Load configuration file
  const result =
    configFile !== undefined
      ? explorer.loadSync(configFile)
      : explorer.searchSync();

  const pluginOptions = {
    ...(result ? result.config : null),
    ...rest,
  };
```

Before this line: https://github.com/callstack/linaria/blob/102397437ff90b35af3089f647968bf528e5481f/src/transform.js#L62

**Comment by satya164** [comment]
> Btw using flow was really a PITA from a contributor experience

We want to migrate to typescript, but it'll take time.

**Comment by Hotell** [comment]
changes applied. Please re-review. cheers

**Comment by satya164** [comment]
Refactored and merged in https://github.com/callstack/linaria/commit/96b82849d8050e1268ac5e93997544e3cee69d09

**Comment by Hotell** [comment]
So why I really appreciate this project and your help, this is not very nice behavior to your contributors. Not at all. 

You should explicitly say next time that you'll do it on your own, rather than leaving someone else spent his free time on something which is gonna be tossed away. Not cool at all.

**Comment by satya164** [comment]
@Hotell I had some time so I thought instead of requesting more changes to your PR, I can finish it quicker and do a release. I based my PR on your PR and also gave attribution to you in my commit. The attribution was mistakenly lost on master because it was squashed instead of "rebase and merge" during merging.

I'm sorry this upset you, but it wasn't my intention.

**Pull request closed without merging**